test/test_sparse.py

```diff
@@ -2363,6 +2364,70 @@ def test_sparse_to_numpy(self):
         t = torch.sparse_coo_tensor(torch.tensor(([0, 0], [2, 0])), torch.tensor([1, 4]))
         self.assertRaises(TypeError, lambda: t.numpy())
 
+    @skipIfRocm
+    def test_hardshrink(self):
+
+        def test_hardshrink_edge_cases(device, dtype):
+            def h(values, l_expected):
+                for l, expected in l_expected.items():
+                    values_tensor = torch.tensor([float(v) for v in values], dtype=dtype, device=device).to_sparse()
+                    expected_tensor = torch.tensor([float(v) for v in expected], dtype=dtype, device=device)
+                    self.assertEqual(expected_tensor == torch.sparse.hardshrink(values_tensor, l).to_dense(),
+                                     torch.ones_like(values_tensor.to_dense(), dtype=torch.bool))
```

Comment:
I am not well-versed in sparse testing. Is there a way to check that the indices are indeed shrunk properly in addition to testing that the dense Tensor is as expected?

Reply:
Sure. Actually looking in detail my new task related to cuda sparse softmax, I figured out that I need to test this function with sparse tensors with a mix of sparse dims. I will do this in the next update.